### PR TITLE
Fix evaluation order for dict operations

### PR DESCRIPTION
**Pull request by sizeoftank**
Fix literals_order issue #1191
not create temporary variables when remapExpr

### CODE REVIEW

**Comment by kmod** [comment]
I don't think this is the best solution here :(  The point of this pass is to un-nest all expressions, and along the way explicitly encode the evaluation order.  So while I think "keeping it nested" might work it's not really what I was hoping for.  There are probably also going to be weird corner cases if you start throwing in other features such as exceptions and control-flow-in-expressions (what if a dict key is something like `1/0 or None`).

Can we have remapDict generate something similar to this:
```
#1 = {}
#2 = <key1>
#3 = <value1>
#1[#2] = #3
#4 = <key2>
#5 = <value2>
#1[#4] = #5
```

**Comment by sizeoftank** [comment]
Thanks for your help, it sounds like good !
I will try do these in few days :-)

**Comment by sizeoftank** [comment]
Hi @kmod 
Please take a look at this if you have the time

File: test/tests/literals_order.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# expected: fail
 # - the CFG expands to the wrong code here
```
Comment (kmod) [inline test/tests/literals_order.py]: Minor, but could you remove this comment too?

**Comment by kmod** [comment]
Looks great!  Would you mind removing that one comment I mentioned and then squashing the two commits together?

**Comment by sizeoftank** [comment]
okay, I updated that :-)

**Comment by kmod** [comment]
Awesome, thanks!

**Pull request merged**